ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -49,6 +50,15 @@
         this.streamingClient = streamingClient;
     }
 
+    public static String generateEngineUriSuggestion(URIBuilder existingEndpoint) {
+        if (!existingEndpoint.getHost().startsWith(IngestClientBase.INGEST_PREFIX)) {
+            throw new IllegalArgumentException("The URI is already an engine URI");
```

Comment:
1. This is specifically a URL, not just a URI
2. It'd be more accurate to say, "The URL is already formatted as the suggested Engine endpoint, so no suggestion can be made"